### ISSUE TITLE
Week 4 Project

### ISSUE THREAD

**Issue by farzaf18** [farzaf18/Week4Project#1]
Overview
The goal of this project is to predict the manner in which they did the exercise. This is the “classe” variable in the training set. This report describes how data was cleaned, how I split “pml-training.csv” into train set and test set, and some of models are investigated.

Exercise
1. Loading add-on package and set seed
set.seed(12345)
library(caret)
2. Download rawdata and submit_data
url_train <- "https://d396qusza40orc.cloudfront.net/predmachlearn/pml-training.csv"
rawdata <- read.csv(url_train, na.strings = c("", "NA"))
url_submit <- "https://d396qusza40orc.cloudfront.net/predmachlearn/pml-testing.csv"
submit_data <- read.csv(url_submit, na.strings = c("", "NA"))
3. Cleaning data
We should delete the column that contains NA to avoid the error. In addition, in order to make accurate predictions, columns that is not related exercise must also be deleted. In particular “X”, “user_name”, “raw_timestamp_part_1”, “raw_timestamp_part_2”, “cvtd_timestamp”, “new_window”, “num_window” are deleted.

#Remove NA cols
colname <- colnames(rawdata)[!colSums(is.na(rawdata)) > 0]
colname
## [58] "magnet_forearm_y"     "magnet_forearm_z"     "classe"
#Slice data related with exercise
colname <- colname[8: length(colname)]
df_wo_NA <- rawdata[colname]

#Check the colnames of df_wo_NA is in submit_data.
#The last colname is "classe"
is.element(colname, colnames(submit_data))
## [45]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE
4. Split data into random train and test
inTrain = createDataPartition(df_wo_NA$classe, p = 3/4)[[1]]
training = df_wo_NA[ inTrain,]
testing = df_wo_NA[-inTrain,]
4. Random Forest
It takes a very long time for training, but it has a high accuracy.

model_rf <- train(classe ~ ., data = training, method = "rf")
pred_rf <- predict(model_rf, testing)
confusionMatrix(testing$classe, pred_rf)
## 
## 
## 
## 
## Balanced Accuracy      0.9968   0.9965   0.9933   0.9965   0.9998
5. Liner Discriminant Analysis
It takes a short time but poor accuracy.

model_lda <- train(classe ~ ., data = training, method = "lda")
pred_lda <- predict(model_lda, testing)
confusionMatrix(testing$classe, pred_lda)
## 
## 
## 
## 
## Balanced Accuracy      0.8544   0.7899   0.7533   0.8029   0.8764
6. Recursive Partitioning and Regression Trees
The results can be confirmed visually, but poor accuracy.

model_rpart <- train(classe ~ ., data = training, method = "rpart")
pred_rpart<- predict(model_rpart, testing)
confusionMatrix(testing$classe, pred_rpart)
## 
## 
## 
## 
## Balanced Accuracy      0.8223   0.6573  0.65625   0.6629   0.8824
library(rattle)
fancyRpartPlot(model_rpart$finalModel)

7. Submit data with Random Forest
We can use the high accuracy model to submit data. In this report the Random Forest accuracy has the highest value 99.45. We can show the prediction.

submit_rf <- predict(model_rf, submit_data)
submit_rf
## Levels: A B C D E
